processing/src/test/java/org/apache/druid/query/UnnestTopNQueryRunnerTest.java

```diff
@@ -0,0 +1,385 @@
+                        new DoubleFirstAggregatorFactory("first", "index", null)
+                    )
+                )
+            )
+        )
+        .postAggregators(QueryRunnerTestHelper.ADD_ROWS_INDEX_CONSTANT)
+        .build();
+
+    List<Result<TopNResultValue>> expectedResults = ImmutableList.of(
+        new Result<>(
+            DateTimes.of("2020-04-02T00:00:00.000Z"),
+            new TopNResultValue(ImmutableList.of())
+        )
+    );
+    assertExpectedResults(expectedResults, query);
+  }
+
+  @Test
+  public void testTopNLexicographicUnnest()
+  {
+    TopNQuery query = new TopNQueryBuilder()
+        .dataSource(QueryRunnerTestHelper.UNNEST_DATA_SOURCE)
+        .granularity(QueryRunnerTestHelper.ALL_GRAN)
+        .dimension(QueryRunnerTestHelper.PLACEMENTISH_DIMENSION_UNNEST)
+        .metric(new DimensionTopNMetricSpec("", StringComparators.LEXICOGRAPHIC))
+        .threshold(4)
+        .intervals(QueryRunnerTestHelper.FIRST_TO_THIRD)
+        .aggregators(commonAggregators)
+        .postAggregators(QueryRunnerTestHelper.ADD_ROWS_INDEX_CONSTANT)
+        .build();
+
+    List<Result<TopNResultValue>> expectedResults = Collections.singletonList(
+        new Result<>(
+            DateTimes.of("2011-04-01T00:00:00.000Z"),
+            new TopNResultValue(
+                Arrays.<Map<String, Object>>asList(
+                    ImmutableMap.of(
+                        QueryRunnerTestHelper.PLACEMENTISH_DIMENSION_UNNEST, "a",
+                        "rows", 2L,
+                        "index", 283.311029D,
+                        "addRowsIndexConstant", 286.311029D,
+                        "uniques", QueryRunnerTestHelper.UNIQUES_1
+                    ),
+                    ImmutableMap.of(
+                        QueryRunnerTestHelper.PLACEMENTISH_DIMENSION_UNNEST, "b",
+                        "rows", 2L,
+                        "index", 231.557367D,
+                        "addRowsIndexConstant", 234.557367D,
+                        "uniques", QueryRunnerTestHelper.UNIQUES_1
+                    ),
+                    ImmutableMap.of(
+                        QueryRunnerTestHelper.PLACEMENTISH_DIMENSION_UNNEST, "e",
+                        "rows", 2L,
+                        "index", 324.763273D,
+                        "addRowsIndexConstant", 327.763273D,
+                        "uniques", QueryRunnerTestHelper.UNIQUES_1
+                    ),
+                    ImmutableMap.of(
+                        QueryRunnerTestHelper.PLACEMENTISH_DIMENSION_UNNEST, "h",
+                        "rows", 2L,
+                        "index", 233.580712D,
+                        "addRowsIndexConstant", 236.580712D,
+                        "uniques", QueryRunnerTestHelper.UNIQUES_1
+                    )
+                )
+            )
+        )
+    );
+    assertExpectedResults(expectedResults, query);
+  }
+
+  @Test
+  public void testTopNStringVirtualColumnUnnest()
```

Comment:
similar comment about virtual column created from multiple columns using `array` in addition to `mv_to_array` test